pkg/collector/resourceutilization/bpf/process_bpf_collector.go

```diff
@@ -92,34 +92,34 @@ func UpdateProcessBPFMetrics(bpfExporter bpf.Exporter, processStats map[uint64]*
 		return
 	}
 	for _, ct := range processesData {
-		comm := C.GoString((*C.char)(unsafe.Pointer(&ct.Command)))
+		comm := C.GoString((*C.char)(unsafe.Pointer(&ct.Comm)))
 
-		if ct.PID != 0 {
+		if ct.Tgid != 0 {
 			klog.V(6).Infof("process %s (pid=%d, cgroup=%d) has %d task clock time %d CPU cycles, %d instructions, %d cache misses, %d page cache hits",
-				comm, ct.PID, ct.CGroupID, ct.TaskClockTime, ct.CPUCycles, ct.CPUInstr, ct.CacheMisses, ct.PageCacheHit)
+				comm, ct.Tgid, ct.CgroupId, ct.TaskClockTime, ct.CpuCycles, ct.CpuInstr, ct.CacheMiss, ct.PageCacheHit)
 		}
 		// skip process without resource utilization
-		if ct.TaskClockTime == 0 && ct.CacheMisses == 0 && ct.PageCacheHit == 0 {
+		if ct.TaskClockTime == 0 && ct.CacheMiss == 0 && ct.PageCacheHit == 0 {
 			continue
 		}
 
 		// if the pid is within a container, it will have a container ID
-		containerID, err := cgroup.GetContainerID(ct.CGroupID, ct.PID, config.EnabledEBPFCgroupID)
+		containerID, err := cgroup.GetContainerID(ct.CgroupId, ct.Tgid, config.EnabledEBPFCgroupID)
```

Comment:
this is a significant change, can we keep it in a separate commit?

Reply:
I could, but I'd have to put in a commit before hand that fixes the (deliberate?) field mixup with the current struct.
Maybe I'll do that anyway (outside of this PR) or we can do it in #1481 which.